examples/fluids/navierstokes.c

```diff
@@ -862,34 +863,40 @@ int main(int argc, char **argv) {
     .time = 0,
   };
 
+  // Create the mesh
   {
     const PetscReal scale[3] = {lx, ly, lz};
     ierr = DMPlexCreateBoxMesh(comm, dim, PETSC_FALSE, NULL, NULL, scale,
                                periodicity, PETSC_TRUE, &dm);
     CHKERRQ(ierr);
   }
-  if (1) {
-    DM               dmDist = NULL;
-    PetscPartitioner part;
-
-    ierr = DMPlexGetPartitioner(dm, &part); CHKERRQ(ierr);
-    ierr = PetscPartitionerSetFromOptions(part); CHKERRQ(ierr);
-    ierr = DMPlexDistribute(dm, 0, NULL, &dmDist); CHKERRQ(ierr);
-    if (dmDist) {
-      ierr = DMDestroy(&dm); CHKERRQ(ierr);
-      dm  = dmDist;
-    }
+
+  // Distribute the mesh over processes
+  DM               dmDist = NULL;
+  PetscPartitioner part;
+
+  ierr = DMPlexGetPartitioner(dm, &part); CHKERRQ(ierr);
+  ierr = PetscPartitionerSetFromOptions(part); CHKERRQ(ierr);
+  ierr = DMPlexDistribute(dm, 0, NULL, &dmDist); CHKERRQ(ierr);
+  if (dmDist) {
+    ierr = DMDestroy(&dm); CHKERRQ(ierr);
+    dm  = dmDist;
```

Comment:
FWIW, the pattern on the left is handy to be explicit that the local variables aren't used outside this scope.

Reply:
Yes, I can add the scope back. That was a slip. I just thought the `if(1)` was unnecessary and I was unclear on why it was there. (it could leave the impression that we want an easy and fast switch to test some features on/off for debugging purposes, but I thought the parallel partitioning of the mesh should be an established feature not to be tested)